mlrun/datastore/targets.py

```diff
@@ -1115,6 +1115,7 @@ def write_dataframe(
             df = self.prepare_spark_df(df)
             df.write.mode("overwrite").save(**options)
         else:
+            df = df.copy(deep=False)
```

Comment:
please add a comment why you are doing a copy.
also, is shallow copy enough?

Reply:
@assaf758  I think that In most cases shallow copy is enough